h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java

```diff
@@ -1537,9 +1535,10 @@ private int run2(String[] args) throws Exception {
     job.setOutputValueClass(Text.class);
 
     FileInputFormat.addInputPath(job, new Path("ignored"));
-    if (outputPath != null) {
+    if (outputPath != null)
       FileOutputFormat.setOutputPath(job, new Path(outputPath));
-    }
+    else
+      job.setOutputFormatClass(NullOutputFormat.class);
```

Comment:
I like this solution. Usually, the user is not interested of hadoop outputs and doesn't want to care about cleaning it as well.